pkg/auth/metadata.go

```diff
@@ -46,108 +43,83 @@ func generateInstallationID(installationID string) string {
-		InstallationIDKeyName)
-	return installationID, err
+	return installationID, nil
 }
 
-func getSetupTimestamp(tx db.Tx) (time.Time, error) {
-	var value string
-	err := tx.GetPrimitive(&value, `SELECT key_value FROM auth_installation_metadata WHERE key_name = $1`,
-		SetupTimestampKeyName)
+func (m *KVMetadataManager) getSetupTimestamp(ctx context.Context) (time.Time, error) {
+	valWithPred, err := m.store.Get(ctx, []byte(model.PartitionKey), []byte(model.MetadataKeyPath(SetupTimestampKeyName)))
 	if err != nil {
 		return time.Time{}, err
 	}
-	return time.Parse(time.RFC3339, value)
+	return time.Parse(time.RFC3339, string(valWithPred.Value))
 }
 
-func writeMetadata(tx db.Tx, items map[string]string) error {
+func (m *KVMetadataManager) writeMetadata(ctx context.Context, items map[string]string) error {
 	for key, value := range items {
-		_, err := tx.Exec(`
-			INSERT INTO auth_installation_metadata (key_name, key_value)
-			VALUES ($1, $2)
-			ON CONFLICT (key_name) DO UPDATE set key_value = $2`,
-			key, value)
+		kvKey := model.MetadataKeyPath(key)
+		err := m.store.Set(ctx, []byte(model.PartitionKey), []byte(kvKey), []byte(value))
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-func (d *DBMetadataManager) UpdateSetupTimestamp(ctx context.Context, ts time.Time) error {
-	_, err := d.db.Transact(ctx, func(tx db.Tx) (interface{}, error) {
-		return nil, writeMetadata(tx, map[string]string{
-			SetupTimestampKeyName: ts.UTC().Format(time.RFC3339),
-		})
-	}, db.WithLogger(logging.Dummy()))
-	return err
+func (m *KVMetadataManager) UpdateSetupTimestamp(ctx context.Context, ts time.Time) error {
+	return m.writeMetadata(ctx, map[string]string{
+		SetupTimestampKeyName: ts.UTC().Format(time.RFC3339),
+	})
 }
 
-func (d *DBMetadataManager) IsInitialized(ctx context.Context) (bool, error) {
-	setupTimestamp, err := d.db.Transact(ctx, func(tx db.Tx) (interface{}, error) {
-		return getSetupTimestamp(tx)
-	}, db.WithLogger(logging.Dummy()), db.ReadOnly())
+func (m *KVMetadataManager) IsInitialized(ctx context.Context) (bool, error) {
+	setupTimestamp, err := m.getSetupTimestamp(ctx)
 	if err != nil {
-		var pgErr *pgconn.PgError
-		if errors.Is(err, pgx.ErrNoRows) ||
-			(errors.As(err, &pgErr) && pgErr.Code == pgerrcode.UndefinedTable) {
+		if errors.Is(err, kv.ErrNotFound) {
 			return false, nil
 		}
-
 		return false, err
 	}
-
-	return !setupTimestamp.(time.Time).IsZero(), nil
+	return !setupTimestamp.IsZero(), nil
 }
 
-func (d *DBMetadataManager) Write(ctx context.Context) (map[string]string, error) {
+func (m *KVMetadataManager) Write(ctx context.Context) (map[string]string, error) {
 	metadata := make(map[string]string)
-	metadata["lakefs_version"] = d.version
+	metadata["lakefs_version"] = m.version
 	metadata["golang_version"] = runtime.Version()
 	metadata["architecture"] = runtime.GOARCH
 	metadata["os"] = runtime.GOOS
-	dbMeta, err := d.db.Metadata(ctx)
```

Comment:
Maybe we need another function to our kv interface? `func Metadata() map[string]string`

It's ok to leave this out as a technical debt, as long as we understand the implications. I would consult with @NoyDavidson2 on what metadata values we stopped collecting to understand the impact.

Reply:
1. Will talk to Noy and update here
2. Nevertheless, I believe that having a relevant `Metadata()` function, is not only a good idea, but somewhat essential. However, we should design something more flexible than what we have today, as our KV is only the mediation layer, and the real data should be undlying-store-specific - I will open an issue for that, once I confirm the needs with Noy